src/lib.rs

```diff
@@ -400,49 +399,55 @@ pub fn check_html_root_url(path: &str, pkg_name: &str, pkg_version: &str) -> Res
+    let krate = syn::parse_crate(&code)
+        .map_err(|source| format!("could not parse {} with source:\n{}", path, source))?;
 
     println!("Checking doc attributes in {}...", path);
-    let mut failed = false;
-    for attr in attrs {
-        if !attr.check_name("doc") {
-            continue;
-        }
-        if let Some(meta_items) = attr.meta_item_list() {
-            for item in meta_items {
-                if let Some(name) = item.name() {
-                    if name != "html_root_url" {
-                        continue;
-                    }
-
-                    let codemap = session.codemap();
-                    let loc = codemap.lookup_char_pos(item.span.lo);
-                    let result =
-                        item.value_str()
-                            .ok_or(String::from("html_root_url attribute without URL"))
-                            .and_then(|url| url_matches(&url.as_str(), pkg_name, &version));
-                    match result {
-                        Ok(_) => println!("{} (line {}) ... ok", path, loc.line),
-                        Err(err) => {
-                            failed = true;
-                            println!("{} (line {}) ... {} in", path, loc.line, err);
-                            if let Ok(snippet) = codemap.span_to_snippet(attr.span) {
-                                println!("{}\n", indent(&snippet));
+    for attr in krate.attrs {
+        match attr {
+            syn::Attribute {
+                style: syn::AttrStyle::Inner,
+                value: syn::MetaItem::List(ref ident, ref nested_meta_items),
+                is_sugared_doc: false,
+            } if ident.as_ref() == "doc" => {
+                for nested_meta_item in nested_meta_items {
+                    let check_result = match *nested_meta_item {
+                        syn::NestedMetaItem::MetaItem(syn::MetaItem::NameValue(ref name, ref value))
+                            if name == "html_root_url" =>
+                        {
+                            match *value {
+                                // accept both cooked and raw strings here
+                                syn::Lit::Str(ref s, _) => url_matches(s, pkg_name, &version),
+                                // non-string html_root_url is probably an error, but we leave
+                                // this check to compiler
+                                _ => continue,
                             }
-                        }
+                        },
+                        syn::NestedMetaItem::MetaItem(syn::MetaItem::Word(ref name))
+                            if name == "html_root_url" =>
+                        {
+                            Err(String::from("html_root_url attribute without URL"))
+                        },
+                        _ => continue,
+                    };
+
+                    match check_result {
+                        Ok(()) => {
+                            // FIXME
```

Comment:
Will you expand this comment to say that we should add line numbers again when the functionality is there?

Reply:
Good idea, I'll do it.